src/libraries/illink-oob.targets

```diff
@@ -0,0 +1,67 @@
+<Project>
+
+  <Target Name="ILLinkTrimOOBAssemblies"
+          AfterTargets="Build"
+          DependsOnTargets="PrepareForAssembliesTrim">
+
+    <Message Text="Trimming $(PackageRID) OOB assemblies with ILLinker..." Importance="high" />
+
+    <PropertyGroup>
+      <LibrariesTrimmedOOBArtifactsPath>$([MSBuild]::NormalizePath('$(ArtifactsBinDir)', 'ILLinkTrimAssembly', '$(BuildSettings)', 'trimmed-oobs'))</LibrariesTrimmedOOBArtifactsPath>
+    </PropertyGroup>
+
+    <ItemGroup>
+
+      <!-- The following is the list of all the OOBs we will ignore for now -->
+      <_OOBsToIgnore Include="System.CodeDom" />
+      <_OOBsToIgnore Include="System.ComponentModel.Composition" />
+      <_OOBsToIgnore Include="System.ComponentModel.Composition.Registration" />
+      <_OOBsToIgnore Include="System.Composition.AttributedModel" />
+      <_OOBsToIgnore Include="System.Composition.Convention" />
+      <_OOBsToIgnore Include="System.Composition.Hosting" />
+      <_OOBsToIgnore Include="System.Composition.Runtime" />
+      <_OOBsToIgnore Include="System.Composition.TypedParts" />
+      <_OOBsToIgnore Include="System.Configuration.ConfigurationManager" />
+      <_OOBsToIgnore Include="System.Speech" />
```

Comment:
I'd include System.DirectoryServices* and System.Management libraries too

Reply:
@marek-safar - can you provide the reasoning? I can imagine a self-contained, trimmed Windows app that would use those libraries.

Note: I don't plan on addressing all the ILLink warnings in the OOB libraries that are being analyzed. My current thinking is we address the warnings in the following libraries for 6.0:

* Microsoft.Extensions.* (used by ASP.NET and Maui)
* System.Drawing.Common (since it is used by WinForms/Wpf)
* System.Memory.Data (since it is new, and should be pretty easy)

The rest of the OOBs that we are analyzing will get their warnings baselined, and at least we can ensure no new warnings will be introduced. As we get feedback, we can address the warnings in those libraries post-6.0.

Thoughts?

Reply:
> can you provide the reasoning?

I think they will have very low usage in .NET6 apps

Baselining should be ok but why make the effort if we don't plan to fix them.

Reply:
> I think they will have very low usage in .NET6 apps

Looks like System.DirectoryServices.dll is in the WinForms/WPF shared framework. Looking on NuGet, the 5.0 versions of these libraries have almost 2M downloads each in about half a year.

> Baselining should be ok but why make the effort if we don't plan to fix them.

It's not that we don't ever plan on fixing them. Just not in the next 2 months.

Also, it provides value because if someone adds new features/APIs to those libraries that are not trim compatible, they will get a build error.